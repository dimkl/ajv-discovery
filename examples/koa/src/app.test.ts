import { app } from "./app";
import * as supertest from "supertest";
import { DiscoverySDKFactory } from "@dimkl/ajv-discovery-sdk";

const server = app.listen();
const request = supertest.agent(server);

describe("Koa app", function () {
  afterAll(() => server.close());

  describe("API", () => {
    test("/discovery returns schemas", async () => {
      const response = await request.get("/discovery");
      expect(response.body).toMatchSnapshot();
    });
  });

  describe("SDK", () => {
    test("discoverySDK using /discovery has customerCreate method", async () => {
      // @ts-ignore
      const { port } = server.address();
      const discoverySDK = (await DiscoverySDKFactory.create({
        apiUrl: `http://localhost:${port}`,
      }));

      // @ts-ignore
      const customerCreateResponse = await discoverySDK.customerCreate({
        email: "dimitris.klouvas@gmail.com",
        password: "1234qwer",
      });

      expect(customerCreateResponse).toEqual({ message: "Customer created" });
    });
  });
});
