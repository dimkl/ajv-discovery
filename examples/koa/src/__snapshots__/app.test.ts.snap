// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Koa app API /discovery returns schemas 1`] = `
{
  "schemas": [
    {
      "$async": true,
      "$id": "/schemas/customer.create",
      "additionalProperties": false,
      "default": {
        "$async": true,
        "$id": "/schemas/customer.create",
        "additionalProperties": false,
        "description": "Customer create schema",
        "http_method": "POST",
        "http_path": "/customers",
        "properties": {
          "email": {
            "format": "email",
            "maxLength": 255,
            "minLength": 1,
            "type": "string",
          },
          "first_name": {
            "maxLength": 255,
            "minLength": 1,
            "type": "string",
          },
          "last_name": {
            "maxLength": 255,
            "minLength": 1,
            "type": "string",
          },
          "password": {
            "maxLength": 255,
            "minLength": 1,
            "type": "string",
          },
        },
        "required": [
          "email",
          "password",
        ],
        "title": "Customer",
        "type": "object",
      },
      "description": "Customer create schema",
      "http_method": "POST",
      "http_path": "/customers",
      "properties": {
        "email": {
          "format": "email",
          "maxLength": 255,
          "minLength": 1,
          "type": "string",
        },
        "first_name": {
          "maxLength": 255,
          "minLength": 1,
          "type": "string",
        },
        "last_name": {
          "maxLength": 255,
          "minLength": 1,
          "type": "string",
        },
        "password": {
          "maxLength": 255,
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "email",
        "password",
      ],
      "title": "Customer",
      "type": "object",
    },
  ],
}
`;
